kolena/_experimental/dataset/_dataset.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import mimetypes
+from dataclasses import asdict
+from enum import Enum
+from typing import Iterator
+from typing import Union
+
+import pandas as pd
+
+from kolena._api.v2.dataset import LoadDatapointsRequest
+from kolena._api.v2.dataset import Path
+from kolena._api.v2.dataset import RegisterRequest
+from kolena._utils import krequests_v2 as krequests
+from kolena._utils.batched_load import _BatchedLoader
+from kolena._utils.batched_load import init_upload
+from kolena._utils.batched_load import upload_data_frame
+from kolena._utils.consts import BatchSize
+from kolena._utils.state import API_V2
+from kolena.errors import InputValidationError
+from kolena.workflow._datatypes import _deserialize_dataobject
+from kolena.workflow._datatypes import _serialize_dataobject
+from kolena.workflow._datatypes import DATA_TYPE_FIELD
+from kolena.workflow._datatypes import TypedDataObject
+
+COL_DATAPOINT = "datapoint"
+TEST_SAMPLE_TYPE = "TEST_SAMPLE"
+FIELD_LOCATOR = "locator"
+FIELD_TEXT = "text"
+
+
+class TestSampleType(str, Enum):
```

Comment:
We should avoid all use of the terminology "test sample" and instead be using "datapoint"